src/Nethermind/Nethermind.Merge.Plugin/Handlers/V1/ForkchoiceUpdatedV1Handler.cs

```diff
@@ -107,11 +111,21 @@ public async Task<ResultWrapper<ForkchoiceUpdatedV1Result>> Handle(ForkchoiceSta
 
             if (!_blockTree.WasProcessed(newHeadBlock.Number, newHeadBlock.GetOrCalculateHash()))
             {
-                _peerRefresher.RefreshPeers(newHeadBlock.Hash, newHeadBlock.ParentHash, forkchoiceState.FinalizedBlockHash);
-                _blockCacheService.SyncingHead = forkchoiceState.HeadBlockHash;
-                _blockCacheService.FinalizedHash = forkchoiceState.FinalizedBlockHash;
-                _mergeSyncController.StopBeaconModeControl();
-                if (_logger.IsInfo) { _logger.Info($"Syncing beacon headers... Request: {requestStr}."); }
+                int processingQueueCount = _processingQueue.Count;
+
+                if (processingQueueCount == 0)
+                {
+                    _peerRefresher.RefreshPeers(newHeadBlock.Hash, newHeadBlock.ParentHash, forkchoiceState.FinalizedBlockHash);
+                    _blockCacheService.SyncingHead = forkchoiceState.HeadBlockHash;
+                    _blockCacheService.FinalizedHash = forkchoiceState.FinalizedBlockHash;
+                    _mergeSyncController.StopBeaconModeControl();
+
+                    if (_logger.IsInfo) { _logger.Info($"Syncing beacon headers... Request: {requestStr}."); }
```

Comment:
hmmm ok, but it still could be something different than "Syncing beacon headers..." I will think about better logging

Reply:
In Beacon sync?